core/mavlink_parameters.cpp

```diff
@@ -324,9 +336,17 @@ void MAVLinkParameters::process_param_ext_value(const mavlink_message_t &message
             if (strncmp(work->param_name.c_str(), param_ext_value.param_id, PARAM_ID_LEN) == 0) {
                 ParamValue value;
                 value.set_from_mavlink_param_ext_value(param_ext_value);
-                _cache[work->param_name] = value;
-                if (work->callback) {
-                    work->callback(true, value);
+                if (value.is_same_type(work->param_value_type)) {
```

Comment:
Here we check that `value.set_from_mavlink_param_ext_value` above did not fail (by silently not updating the value), right?

But then I don't get why the error is called "WRONG_TYPE" :thinking:

Reply:
When you do the call to get a param you need to specify which type you expect, and that's what we compare here.
So one type comes from the mavlink message and one from the work item from the user.

Reply:
I guess I'll need to learn about how the protocol works at some point :sweat_smile:.